internal/backend/memory.go

```diff
@@ -8,117 +8,216 @@ import (
 	"errors"
 	"fmt"
 	"sync"
+	"time"
 )
 
 // StorageMemory is an implementation of the backend storage that stores data in memory
 type StorageMemory struct {
-	buckets map[string][]Object
+	buckets map[string]bucketInMemory
 	mtx     sync.RWMutex
 }
 
+type bucketInMemory struct {
```

Comment:
now, rather than a bucket being represented by a name (and implemented just as a list of objects associated to the name), we promote the Bucket entity (so props can be attached there) and the in-memory implementation extends the struct to store lists of objects.